### DISCUSSION TITLE
How does Hot Module Reload swap in changed files?

### DISCUSSION THREAD

**Discussion by marialovesbeans** [webpack/webpack#13059] [Q&A]
Hey guys, does anyone understand how Webpacks HMR works under-the-hood (for example for React Fast Refresh)? i.e. How is it able to swap out a modified module at runtime?

Precisely, after browser fetches the new changed file, how does it replace the old module, and have the new code take into effect?

**Comment by alexander-akait** [comment]
Here docs https://webpack.js.org/concepts/hot-module-replacement/, what do you don't understand?
